2023/02/how-to-create-a-hugo-website-without-a-theme.md

```diff
@@ -0,0 +1,340 @@
+```
+
+You may notice a `katex: true` key & value in the front matter. That's an optional extra goody I've added, which we'll go over soon!
+
+### Building and using the development server
+
+To deploy the site to the `build` directory, run:
+
+```plain
+$ hugo
+```
+
+Or to use Hugo's development server (one of its strongest features, in my opinion), go to the base site directory (`notes`) and run:
+
+```plain
+$ hugo serve
+```
+
+It will give you a localhost address with a port (`1313` unless it's taken) that you can open in your browser of choice.
+
+Now you should have a shiny new Hugo site to work with, no theme required!
+
+Here's the list page template in action:
+
+![A web browser open to localhost:39333, showing the home page of the notes site: A header reads "Notes", followed by smaller headers. The first is pink, reading "Winter-2022", with its own bulleted list containing the item "Lecture 9", also in pink. The second is black and reads "Other", with no list beneath it.](/blog/2023/02/how-to-create-a-hugo-website-without-a-theme/home.webp)
+
+And the single page template, plus our content file:
+
+![A web browser open to localhost:39333/winter-2022/cs-235/lecture-9/. At the top of the page is pink text (a link) reading "Back to Winter-2022". Below is a header reading "Lecture 9". Followed by the text of the lecture-9.md file. sumto() appears in an inline code block, as well as the later `sumto()...` text.](/blog/2023/02/how-to-create-a-hugo-website-without-a-theme/lecture-9.webp)
+
+### Adding KaTeX support
+
+[KaTeX](https://katex.org/) is a lightweight JavaScript TeX implementation. I use enough math in my notes as an engineering student that I wanted an easy way to show math notation.
+
+Although I use math notation pretty often, I wanted to only load the JavaScript when necessary. For this, I set a front matter variable (`katex`), and render a `katex.html` partial in the site's `<head>` if it's there.
+
+Add the following to `baseof.html`, in the `<head>` tag:
+
+```plain
+{{ if .Params.katex }}{{ partial "partials/katex.html" . }}{{ end }}
+```
+
+The `partial` function takes a path relative to `layouts`, so `partials/katex.html` finds the file in `layouts/partials/katex.html`.
+
+Put the following in `layouts/partials/katex.html`:
+
+```plain
+<link rel="stylesheet" href="/katex/katex.min.css">
+<script defer src="/katex/katex.min.js"></script>
+<script defer src="/katex/contrib/auto-render.min.js" onload="renderMathInElement(document.body);"></script>
+```
+
+I [downloaded](https://katex.org/docs/browser.html#download--host-things-yourself) KaTeX myself, but you could easily modify this partial to use CDN-hosted files.
+
+If you host the files yourself, like I did, just download and move the necessary files to `static/katex`. You only need to keep a few files out of the ones KaTeX ships with:
+
+```plain
+└── static
+    └── katex
+        ├── contrib
+        │   └── auto-render.min.js
+        ├── fonts
+        │   └── ✀
```

Comment:
Is there a literal file `✀` here? If not, I fear this cutting out all the fonts may be more confusing than helpful.